Gems/AudioSystem/Code/Source/Engine/ATLAudioObject.cpp

```diff
@@ -594,18 +594,22 @@ namespace Audio
         // warn if there is activity on an object being cleared
         if (!m_cActiveEvents.empty())
         {
+            AZ_PUSH_DISABLE_WARNING_GCC("-Wdangling-pointer");
```

Comment:
The call to GetEventIDs, should remove the `.c_str()` call and assign the result to an AZStd::string.
```c++
const AZStd::string sEventString = GetEventIDs("; ");
```